src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js

```diff
@@ -71,9 +71,10 @@ class BaseDrawerNavigator extends Component {
                     <Drawer.Screen
                         key={screen.name}
                         name={screen.name}
-                        component={screen.component}
```

Comment:
Pretty sure this is the recommended practice for rendering a screen component. Why are we changing it?

Reply:
@mdneyazahmad back to drawing board ig 😅

Reply:
source: https://reactnavigation.org/docs/screen/#children

@mdneyazahmad makes sense. Thanks for the context

Reply:
I'm kinda lost here. Why does this all need to happen at such a low level and why are we duplicating Onyx data in the `MainDrawerNavigator` `state` instead of just referencing the value that is stored in Onyx and have the SidebarLinks update when it changes?

Reply:
The summary of changes I'm seeing are:

- Change name of `currentlyViewedReportID` (doesn't seem necessary)
- Stop passing `screen.component` here because we are using a children prop (I feel like this is very bad for performance)
- Storing the `currentlyViewReportID` in `MainDrawerNavigator` state is think so we can set the value from inside the `ReportScreen` but isn't that what `updateCurrentlyViewedReportID()` was doing before?

Can we just rewind and answer the question about why `updateCurrentlyViewedReportID()` does not update the value in `SidebarScreen`?

Reply:
Just add it to Onyx here if we need that value?

https://github.com/Expensify/App/blob/0609796c7e0d3768217bc381b7320420c41b27d6/src/pages/home/sidebar/SidebarScreen.js#L194

Reply:
> Can we just rewind and answer the question about why updateCurrentlyViewedReportID() does not update the value in SidebarScreen?

When we opens a report, this updates `currentlyViewReportID` in `Onyx`. The issue here is that when user open multiple report in different tabs. `Onyx` will contain the latest report id as `currentlyViewReportID`. We are subscribing to this variable in `SidebarScreen` therefore the highlighted links in the left sidebar is not synced with the report on multiple tab. As soon the user changes the report, it will updated the variable in `Onyx` and every other tabs highlighted links will change.

> Change name of currentlyViewedReportID (doesn't seem necessary)

It works without changing this name but `lastViewedReportID` makes much sense here. User can open multiple reports but it will contain the id of last report opened.

Reply:
@marcaaron this is the problem that we're trying to solve

1. Open _report A_ in tab 1
2. Open _report B_ in tab 2
3. Check the highlighted report in LHN of tab 1

Expected result
Tab 1 has _report A_ highlighted.

Actual Result
Tab 1 has _report B_ highlighted.

This is because both tabs are using `currentlyViewedReportID` from Onyx to know which report to highlight.

In simple words, currentlyViewedReportID in Onyx is like a global variable having the **same** value between tabs, but we want a variable which has different value across tabs.

Reply:
Just going to restate the problem and offer some ideas...

**Problem:**

- Onyx notifies other tabs that a value has changed by default
- There are cases where we don't want this to happen 
- e.g. updating the active report will update other tabs and we update the active report without navigating to it

**Possible Solutions:**

1. Move the variable out of Onyx completely
2. Keep it in Onyx and in the state somewhere (the chosen solution)
3. Disable Onyx notifying other tabs when a value changes for a certain key

Here's how I think the last one might work...

1. Add a `keysToDisableSyncEvents` array to `Onyx.init()` options
2. Pass this to `keepInstancesSync` [here](https://github.com/Expensify/react-native-onyx/blob/2d42566779884bb445aa59dd48b5bcdff6c0a4e6/lib/Onyx.js#L843)
3. Add [here](https://github.com/Expensify/react-native-onyx/blob/2d42566779884bb445aa59dd48b5bcdff6c0a4e6/lib/storage/WebStorage.js#L43) `if (_.contains(keysToDisableSyncEvents, onyxKey)) return;`

Reply:
>It works without changing this name but lastViewedReportID makes much sense here. User can open multiple reports but it will contain the id of last report opened.

Please do not make this change, thanks!

Reply:
I think solution 3 is a good one. It might be needed in the future. Onyx is always the easiest way to manage updates.

Reply:
Thanks for the solution 3, that is exactly what we need.
@mdneyazahmad let's get it done! :)

Reply:
We need only this change

https://github.com/Expensify/App/blob/43910fe5c20584b05fe56013005e8e59ba93e8a6/src/setup/index.js#L24-L26

```diff
  captureMetrics: Metrics.canCaptureOnyxMetrics(),
+ keysToDisableSyncEvents: [ONYXKEYS.CURRENTLY_VIEWED_REPORTID],
```


I will create a pr to `react-native-onyx` to add the feature as suggested my @marcaaron